extensions/amp-jwplayer/0.1/amp-jwplayer.js

```diff
@@ -167,11 +449,123 @@ class AmpJWPlayer extends AMP.BaseElement {
     }
     return placeholder;
   }
+
+  /**
+   * @param {Object} data
+   * @private
+   */
+  onReady_(data) {
+    const {element} = this;
+
+    this.playlistItem_ = {...data.playlistItem};
+    this.muted_ = !!data.muted;
+    this.playerReadyResolver_(this.iframe_);
+    element.dispatchCustomEvent(VideoEvents.LOAD);
+  }
+
+  /**
+   * @param {Event} messageEvent
+   * @private
+   */
+  onMessage_(messageEvent) {
+    if (
+      !this.iframe_ ||
+      !messageEvent ||
+      messageEvent.source != this.iframe_.contentWindow
+    ) {
+      return;
+    }
+
+    const messageData = getData(messageEvent);
+
+    if (!isJsonOrObj(messageData)) {
+      return;
+    }
+
+    const data = objOrParseJson(messageData);
+    const event = data['event'];
+    const detail = data['detail'];
+
+    // Log any valid events
+    dev().info('JWPLAYER', 'EVENT:', event || 'anon event', detail || data);
+
+    if (event === 'ready') {
+      detail && this.onReadyOnce_(detail);
+      return;
+    }
+
+    const {element} = this;
+
+    if (redispatch(element, event, JWPLAYER_EVENTS)) {
+      return;
+    }
+
+    if (detail && event && eventHandlers[event]) {
+      eventHandlers[event](detail, this);
+    }
+  }
+
+  /**
+   * @param {string} method
+   * @param {number|boolean|string|Object|undefined} [optParams]
+   * @private
+   */
+  sendCommand_(method, optParams) {
+    this.playerReadyPromise_.then(() => {
+      if (!this.iframe_ || !this.iframe_.contentWindow) {
+        return;
+      }
+
+      dev().info('JWPLAYER', 'COMMAND:', method, optParams);
+
+      this.iframe_.contentWindow./*OK*/ postMessage(
+        JSON.stringify(
+          dict({
+            'method': method,
+            'optParams': optParams,
+          })
+        ),
+        '*'
+      );
+    });
+  }
+
+  /**
+   * @private
+   * @return {boolean}
+   */
+  isSafariOrIos_() {
+    const platform = Services.platformFor(this.win);
+
+    return platform.isSafari() || platform.isIos();
+  }
+
+  /**
+   * @private
+   * @return {string}
+   */
+  getSingleLineEmbed_() {
+    const isDev = getMode(this.win).localDev;
+    const cid = encodeURIComponent(this.contentid_);
+    const pid = encodeURIComponent(this.playerid_);
+    let baseUrl = `https://content.jwplatform.com/players/${cid}-${pid}.html`;
+
+    if (isDev) {
```

Comment:
Since this only works on the local development server, could we remove it?

Reply:
It's used extensively for dev testing against various versions of the player. Is it harmful to leave this in?

Reply:
I'm unsure why this is necessary, why would this need to be done from the AMP dev's side?

Reply:
While the way this is normally set up is through our production dashboard, we want to be able to test local/dev builds of the player which requires a different url structure than the production environment. 

So if it is dev, and no test_page attribute is found on the page url, it will set up the traditional way, but if the attribute is found, we can overwrite with the given player.